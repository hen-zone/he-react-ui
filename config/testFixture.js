import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
