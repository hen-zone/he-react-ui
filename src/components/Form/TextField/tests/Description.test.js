import fixture from '../fixtures/Description.fixture.js'
import testFixture from '../../../../../config/testFixture'

testFixture(fixture, 'Description')
