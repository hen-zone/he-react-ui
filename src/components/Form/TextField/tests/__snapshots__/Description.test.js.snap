// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> rendered correctly with Description fixture 1`] = `
<TextField
  description="Optional descriptive label"
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  label="First Name"
  marker={false}
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  value=""
>
  <div
    className="outer"
  >
    <div
      className="block"
    >
      <label
        className="label"
        htmlFor="demo"
      >
        First Name
      </label>
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
    <label
      className="description"
      htmlFor="demo"
    >
      Optional descriptive label
    </label>
  </div>
</TextField>
`;
